packages/entities/entities-plugins/src/components/PluginSelect.vue

```diff
@@ -0,0 +1,497 @@
+
+  if (props.config.app === 'konnect') {
+    url = url.replace(/{controlPlaneId}/gi, props.config?.controlPlaneId || '')
+  } else if (props.config.app === 'kongManager') {
+    url = url.replace(/\/{workspace}/gi, props.config?.workspace ? `/${props.config.workspace}` : '')
+  }
+
+  return url
+})
+
+const fetchEntityPluginsUrl = computed((): string => {
+  if (props.config?.entityType && props.config?.entityId) {
+    let url = `${props.config.apiBaseUrl}${endpoints.list[props.config.app].forEntity}`
+
+    if (props.config.app === 'konnect') {
+      url = url.replace(/{controlPlaneId}/gi, props.config?.controlPlaneId || '')
+    } else if (props.config.app === 'kongManager') {
+      url = url.replace(/\/{workspace}/gi, props.config?.workspace ? `/${props.config.workspace}` : '')
+    }
+
+    return url
+      .replace(/{entityType}/gi, props.config.entityType || '')
+      .replace(/{entityId}/gi, props.config.entityId || '')
+  }
+
+  return ''
+})
+
+// race condition between fetch of available plugins and setting
+// disabled/ignored plugins
+// if disabled/ignored plugins changes before availablePlugins fetch
+// wait for fetch to finish and don't attempt to build plugin list yet
+// if disabled/ignored plugins changes AFTER availablePlugins fetch
+// rebuild the list
+watch(() => props.disabledPlugins, (val, oldVal) => {
+  if (!objectsAreEqual(val, oldVal) && !isLoading.value) {
+    pluginsList.value = buildPluginList()
+  }
+})
+
+watch(() => props.ignoredPlugins, (val, oldVal) => {
+  if (!objectsAreEqual(val, oldVal) && !isLoading.value) {
+    pluginsList.value = buildPluginList()
+  }
+})
+
+const userCanCreateCustom = ref(false)
+const userCanEditCustom = ref(false)
+const userCanDeleteCustom = ref(false)
+onBeforeMount(async () => {
+  // Evaluate the user permissions
+  userCanCreateCustom.value = await props.canCreateCustom()
+  userCanEditCustom.value = await props.canEditCustom()
+  userCanDeleteCustom.value = await props.canDeleteCustom()
+})
+
+onMounted(async () => {
+  isLoading.value = true
+  hasError.value = false
+  fetchErrorMessage.value = ''
+
+  emit('loading', isLoading.value)
+
+  try {
+    const res = await axiosInstance.get(availablePluginsUrl.value)
+
+    // TODO: endpoints temporarily return different formats
+    if (props.config.app === 'konnect') {
+      const { names: available } = res.data
+      availablePlugins.value = available || []
+    } else if (props.config.app === 'kongManager') {
+      const { plugins: { available_on_server: aPlugins } } = res.data
+      availablePlugins.value = aPlugins ? Object.keys(aPlugins) : []
+    }
+  } catch (error: any) {
+    hasError.value = true
+    fetchErrorMessage.value = getMessageFromError(error)
+  }
+
+  // fetch scoped entity to check for pre-existing plugins
+  if (fetchEntityPluginsUrl.value) {
+    try {
+      const res = await axiosInstance.get(fetchEntityPluginsUrl.value)
+      const { data } = res.data
+
+      if (data?.length) {
```

Comment:
If you destructure `data` above you won't need the optional chaining here

Reply:
This one is `data.data`, wouldn't I still need to check that it's defined and has a non-zero length?

Reply:
If it's `data.data` then you can do this:

```ts
const { data: { data } } = await axiosInstance.get(fetchEntityPluginsUrl.value)
```

Reply:
Right, but wouldn't I still need the same conditional? This just means I can remove line `450` right? 
![image](https://github.com/Kong/public-ui-components/assets/67973710/86364619-356b-4f0d-a254-9626f96bbde3)

Reply:
yes